### DISCUSSION TITLE
add_requireconfs 指定版本是否总是需要 override

### DISCUSSION THREAD

**Discussion by cyfdecyf** [xmake-io/xmake#2145] [Q&A] [answered]
官网 [`add_requireconfs`](https://xmake.io/#/zh-cn/manual/global_interfaces?id=add_requireconfs) 文档有个例子：

> 如果 add_requires("zlib 1.2.11") 中设置了版本，就会优先使用 add_requires 的配置，完全忽略 add_requireconfs 里面的版本配置，当然我们也可以通过 override 来完全重写 add_requires 中指定的版本。
>
>```lua
>add_requires("zlib 1.2.11")
>add_requireconfs("zlib", {override = true, version = "1.2.10"})
>```

若代码如下：

```lua
add_requires("zlib")
add_requireconfs("zlib", {override = true, version = "1.2.10"})
```

`add_requires` 并未指定版本，测试发现 `add_requireconfs` 中指定的版本也被忽略。是否因为不指定版本隐含指定最新版？

---

我想使用一个 common.lua 脚本来固定 xrepo 安装的所有包的版本，代码如下：

```lua
function gflags_configs()
    return {version = "v2.2.2", system = false, configs = {mt = true, shared = false}}
end

function glog_configs()
    return {version = "v0.5.0", system = false, configs = {gflags = true, unwind = true, shared = false}}
end

function libunwind_configs()
    return {version = "v1.5", system = false}
end
```

具体某个包的文件，以 glog.lua 为例：

```lua
includes("common.lua")
add_requires("glog", glog_configs())
add_requireconfs("glog.gflags", gflags_configs())
add_requireconfs("glog.libunwind", libunwind_configs())
```

为了固定版本，为每个包都指定 `override = true` 能够工作。想请教是否有更好的做法？

**Comment by waruqi** [comment, marked as answer]
要么在 add_requires 里指定版本，add_requires("xxx 1.0") 要么在 add_requireconfs + override 改写版本，默认走的 latest 拉取的最新版本
